application-task-ui/src/main/resources/TaskManager/GanttMacro/Macro.xml

```diff
@@ -0,0 +1,2511 @@
+ */
+define('taskgantt-override-functions', [
+  'jquery',
+  'taskgantt-utils',
+  'taskgantt-ui',
+  'taskgantt-service-urls',
+], function ($, taskutil, taskui, taskurls) {
+  /**
+   * Custom implementation of trigger_event from the frappe-gantt(v0.6.1) library.
+   * Bind the parent gantt object to 'this' in the event handlers.
+   */
+  function custom_trigger_event(event, args) {
+    if (this.options['on_' + event]) {
+      this.options['on_' + event].apply(this, args);
+    }
+  }
+  /**
+   * Custom implementation of render from the frappe-gantt(v0.6.1) library.
+   * Uses the original render function from the frappe-gantt library.
+   * Adds a custom popup and the custom svg markup for task bar fade on no due date.
+   */
+  function custom_render() {
+    const result = this._render(); // Use the default render implementation first.
+    taskui.addSvgStyles(this.$svg); // SVG specific style definitions, with no css equivalent.
+    taskui.setupProgressPopup(this);
+    return result;
+  };
+  /**
+   * Custom implementation of get_date_info from the frappe-gantt(v0.6.1) library.
+   * Fixes the hour text display in Quarter Day view mode.
+   * In the default frappe-gantt implementation, the header text is offset by 12h.
+   * Example: If a task starts at 06:00, it appears under the 18:00 section in the UI.
+   */
+  function custom_get_date_info(date, last_date, i) {
+    let result = this._get_date_info(date, last_date, i);
+    const base_pos = {
+      x: i * this.options.column_width,
+      lower_y: this.options.header_height,
+      upper_y: this.options.header_height - 25,
+    };
+
+    const x_pos = {
+      'Quarter Day_lower': (this.options.column_width * 4),
+      'Quarter Day_upper': (this.options.column_width * 4) / 2,
+      'Half Day_lower': (this.options.column_width * 2),
+      'Half Day_upper': (this.options.column_width * 2) / 2,
+      Day_lower: this.options.column_width / 2,
+      Day_upper: (this.options.column_width * 30) / 2,
+      Week_lower: 0,
+      Week_upper: (this.options.column_width * 4) / 2,
+      Month_lower: this.options.column_width * 1.5,
+      Month_upper: (this.options.column_width * 12) / 2,
+      Year_lower: this.options.column_width / 2,
+      Year_upper: (this.options.column_width * 30) / 2,
+    };
+    const tweakedPos = {
+      upper_x: base_pos.x + x_pos[`${this.options.view_mode}_upper`],
+      upper_y: base_pos.upper_y,
+      lower_x: base_pos.x + x_pos[`${this.options.view_mode}_lower`],
+      lower_y: base_pos.lower_y,
+    };
+    Object.assign(result, tweakedPos);
+    return result;
+  };
+  /**
+   * Custom implementation of bind_bar_events from the frappe-gantt(v0.6.1) library.
+   * Don't move dependencies when moving a task.
+   * This prevents unintentional updates and more freedom when moving tasks.
+   */
+  function custom_bind_bar_events() {
+    // Redefine some jQuery functions so they work on our SVG elements.
```

Comment:
The Gantt library redefines some jQuery functions, probably in order to remain 'dependency free'.

I'll have to test if jQuery has any edge cases which come into play here, like the jQuery [on()](https://api.jquery.com/on/) function has this:
```Note: Delegated event handlers do not work for SVG.```

Reply:
Since this method is copied from frappe-gantt and has only some things changed, there should be comments added for the exact thing that were changed. This will make future upgrades easier